File: 2419.longest-subarray-with-maximum-bitwise-and/solution.go
func longestSubarray(nums []int) int {
    maxBitwise := 0
    maxLength := 0
    curLength := 0
    for _, n := range nums {
        switch {
        case n > maxBitwise:
            maxBitwise = n
            maxLength = 1
            curLength = 1
        case n == maxBitwise:
            curLength++
            if curLength > maxLength {
                maxLength = curLength
            }
        default:
            curLength = 0
        }
    }
    return maxLength
}

File: 2419.longest-subarray-with-maximum-bitwise-and/solution.kt
class Solution {
    fun longestSubarray(nums: IntArray): Int {
        val (maxLength, _, _) = nums.fold(Triple(0, 0, 0)) { (maxLength, curLength, maxBitwise), num ->
            when {
                num > maxBitwise -> Triple(1, 1, num)
                num == maxBitwise -> Triple(maxOf(curLength + 1, maxLength), curLength + 1, maxBitwise)
                else -> Triple(maxLength, 0, maxBitwise)
            }
        }
        return maxLength
    }
}